sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -3455,6 +3455,11 @@ object functions {
 
   /**
    * Returns an array of elements for which a predicate holds in a given array.
+   * {{{
+   *   df.selectExpr("filter(col, x -> x % 2 == 0)")
```

Comment:
@gatorsmile, can we use the actual `filter` API in this example?

Reply:
Yes. I added another example using the filter API directly